### PR TITLE
Feature/Add configurable endpoints

### PR DESCRIPTION
**Pull request by simonrauch**
**TLDR;**

- added configurable endpoinnt
- added test
- updated composer dependencies
- updated docs and changelog

**What to test & how to test**

- add .env file with `EVENTSERVER_TOKEN` set
- run tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## unreleased
```
Comment (SahinU88) [inline CHANGELOG.md]: should be `[unreleased]`

File: composer.json [hunk 2]
```diff
@@ -15,11 +15,11 @@
         "guzzlehttp/guzzle": "^6.3"
     },
     "require-dev": {
-        "phpunit/phpunit": "^7.2",
-        "codedungeon/phpunit-result-printer": "^0.19.10",
-        "nunomaduro/collision": "^2.0",
+        "phpunit/phpunit": "^8.5",
```
Comment (SahinU88) [inline composer.json]: we could even go to `^9.0.0`

File: composer.json [hunk 3]
```diff
@@ -15,11 +15,11 @@
...
+        "codedungeon/phpunit-result-printer": "^0.27.0",
+        "nunomaduro/collision": "^3.0",
```
Comment (SahinU88) [inline composer.json]: we could even go to `^4.1.3`

**Review by SahinU88 — CHANGES_REQUESTED** [review]
For the moment, the withEndpoint is ok. We should add a repsitory pattern maybe, so we could use the library in general somethink like:

```
Eventserver::events()->get()...
Eventserver::categories()->get()..

// or with Repository classes
Events::get()...
Categories::get()..
```

but this is a feature change and not part of this review :)

File: CHANGELOG.md (hunk 1 above)
Comment (simonrauch) [reply CHANGELOG.md]: 👍

File: composer.json (hunk 2 above)
Comment (simonrauch) [reply composer.json]: 👍

File: composer.json (hunk 3 above)
Comment (simonrauch) [reply composer.json]: updating this dependency seems to break something within phpunit:

```
> vendor/bin/phpunit
Class "NunoMaduro\Collision\Adapters\Phpunit\Listener" does not exist
Script vendor/bin/phpunit handling the test event returned with error code 2
```

**Comment by simonrauch** [comment]
> For the moment, the withEndpoint is ok. We should add a repsitory pattern maybe, so we could use the library in general somethink like:
> 
> ```
> Eventserver::events()->get()...
> Eventserver::categories()->get()..
> 
> // or with Repository classes
> Events::get()...
> Categories::get()..
> ```
> 
> but this is a feature change and not part of this review :)

Yup i thought so too, but i was not sure if this project is used somewhere already and i did not want to break backwards compatibility.

File: composer.json (hunk 3 above)
Comment (SahinU88) [reply composer.json]: I saw that we had to update the php-unit file. I just commited it. 👍

Comment (SahinU88) [reply composer.json]: sry, i forgot that it's a for from your side ^^

here would be the new config-file:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<phpunit bootstrap="vendor/autoload.php" backupGlobals="false" backupStaticAttributes="false" colors="true"
         verbose="true" convertErrorsToExceptions="true" convertNoticesToExceptions="true"
         convertWarningsToExceptions="true" processIsolation="false" stopOnFailure="false"
         printerClass="NunoMaduro\Collision\Adapters\Phpunit\Printer">
    <testsuites>
        <testsuite name="Towa Test Suite">
            <directory>tests</directory>
        </testsuite>
    </testsuites>
    <filter>
        <whitelist>
            <directory suffix=".php">src/</directory>
        </whitelist>
    </filter>
</phpunit>
```

**Review by SahinU88 — CHANGES_REQUESTED** [review]
Added the xml-config file. After integrating that, we can merge it :)

**Review by SahinU88 — APPROVED** [review]

**Pull request merged**